File: 01b.js
import * as fs from 'fs';

const filename = "01.dat";
fs.readFile(filename, (err, data) => {
    if (err) throw err;

    var buffer = [];
    var buffer_sum = 0;
    var previous_sum;
    var increased = 0;

    data.toString().split("\n").forEach(line => {
        buffer.push(parseInt(line));
        if (buffer.length >= 3) {
            if (buffer.length == 4)
                buffer.shift();
            buffer_sum = buffer.reduce((sum, item) => sum + item);
            if (previous_sum !== undefined && buffer_sum > previous_sum)
                increased++;
            previous_sum = buffer_sum;
        }
	});

    console.log(increased);
});

File: 01b.py
#!/usr/bin/python3

with open('01.dat') as f:
    data = f.read()
measurements = data.split('\n')
measurements.pop()

buffer = []
buffer_sum = 0
previous_sum = None
increased = 0
for measurement in measurements:
    buffer.append(int(measurement))
    if (len(buffer) < 3):
        continue
    if (len(buffer) == 4):
        del buffer[0]
    buffer_sum = sum(buffer)
    if (previous_sum != None and buffer_sum > previous_sum):
        increased += 1
    previous_sum = buffer_sum
print(increased)